File: framework/test2/void/revstr/revstr.go
package main

import (
    . "missingjs.com/soda/leetcode"
    "missingjs.com/soda/unittest"
    "missingjs.com/soda/util"
)

var logger = util.Logger
var _used ListNode

func reverseString(s []byte) {
    i, j := 0, len(s) - 1
    for i < j {
        temp := s[i]
        s[i] = s[j]
        s[j] = temp
        i++
        j--
    }
}

func main() {
    // create tester by function
    work := unittest.CreateWork(reverseString)
    // OR create by struct factory
    // work := unittest.CreateWorkForStruct(Constructor)
    // work.SetValidator(func(e,r)bool)
    work.CompareSerial = true
    work.Run()
}

File: framework/test2/void/revstr/Revstr.cs
using Soda.Unittest;
using Soda.Leetcode;

public class Solution {
    public void ReverseString(char[] s)
    {
        int i = 0, j = s.Length - 1;
        while (i < j) {
            (s[i], s[j]) = (s[j], s[i]);
            ++i;
            --j;
        }
    }
}

public class Revstr
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().ReverseString));
        // var work = WorkFactory.ForStruct<STRUCT>();
        // work.SetValidator((e, r) => ...);
        work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.ReadStdin()));
    }
}
